gearcmd/worker.go

```diff
@@ -43,6 +44,11 @@ func (conf TaskConfig) Process(job baseworker.Job) ([]byte, error) {
 		data["error_message"] = err.Error()
 		data["value"] = 0
 		data["success"] = false
+		if conf.RetryCount > 0 {
+			log.Println(kayvee.FormatLog("gearcmd", kayvee.Error, "RETRY", data))
```

Comment:
I refactored the retry code to use a loop instead of recursion, and I think it ended up simpler.  It also fixed the logging problem.  Now the logging looks like:

- START -  The info message when a job first starts.
- RETRY - The error message when a job fails, but is being retried.
- RESTART - The info message when a job restarts.
- END - The error or info message when a job ends.

I can change these to be something different if they're not clear.  Thoughts?